Description: add pam security verify
Author: huxd1532 <huxd1532@163.com> 闫博文 <yanbowen@deepin.com>

Origin: https://gerrit.uniontech.com/plugins/gitiles/base/sudo/+/9bba3fe3c65248a18db6db6e08b650710cce1268
Bug: <url in upstream bugtracker>
Last-Update: 2022-05-19

--- sudo-1.9.10.orig/plugins/sudoers/auth/pam.c
+++ sudo-1.9.10/plugins/sudoers/auth/pam.c
@@ -286,8 +286,9 @@ int
 sudo_pam_verify(struct passwd *pw, char *prompt, sudo_auth *auth, struct sudo_conv_callback *callback)
 {
 	const char *envccname;
-    const char *s;
+    const char *s, *pam_item;
     int *pam_status = (int *) auth->data;
+    int rc;
     debug_decl(sudo_pam_verify, SUDOERS_DEBUG_AUTH);
 
     def_prompt = prompt;	/* for converse */
@@ -332,6 +333,12 @@ sudo_pam_verify(struct passwd *pw, char
 	    sudo_debug_printf(SUDO_DEBUG_WARN|SUDO_DEBUG_LINENO,
 		"pam_authenticate: %d", *pam_status);
 	    debug_return_int(AUTH_FAILURE);
+	case PAM_ABORT:
+	    rc = pam_get_item(pamh, PAM_SERVICE, &pam_item);
+	    if((rc == PAM_SUCCESS)&&(!strcmp(pam_item, "security-verify"))){
+		log_warningx(0, N_("No root privileges. Please request root access in developer mode in Control Center."));
+		debug_return_int(AUTH_FATAL);
+	    }
 	default:
 	    s = sudo_pam_strerror(pamh, *pam_status);
 	    log_warningx(0, N_("PAM authentication error: %s"), s);
--- sudo-1.9.10.orig/plugins/sudoers/po/zh_CN.po
+++ sudo-1.9.10/plugins/sudoers/po/zh_CN.po
@@ -1465,6 +1465,10 @@ msgstr "%s：无法验证目标！可能
 msgid "unable to initialize PAM: %s"
 msgstr "无法初始化 PAM：%s"
 
+#: plugins/sudoers/auth/pam.c:342
+msgid "No root privileges. Please request root access in developer mode in Control Center."
+msgstr "无root权限，如想获得root权限可以在控制中心选择进入开发者模式"
+
 #: plugins/sudoers/auth/pam.c:337
 #, c-format
 msgid "PAM authentication error: %s"
