Description: Fix endian detect for sunway
Author: Miao Changwei <miaochangwei@uniontech.com>

---
Last-Update: 2022-04-22

--- a/include/compat/endian.h
+++ b/include/compat/endian.h
@@ -57,7 +57,7 @@
        defined(__ns3200) || defined(sun386) || defined(vax) || \
        defined(__vax) || defined(__x86__) || defined(__riscv) || \
        (defined(sun) && defined(__powerpc)) || \
-       (!defined(__hpux) && defined(__ia64))
+       (!defined(__hpux) && defined(__ia64)) || defined(__sw_64__)
 #  define BYTE_ORDER	LITTLE_ENDIAN
 # elif defined(__68k__) || defined(apollo) || defined(BIT_ZERO_ON_LEFT) || \
        defined(__convex__) || defined(_CRAY) || defined(DGUX) || \
